src/execution/operator/join/physical_nested_loop_join.cpp

```diff
@@ -284,8 +284,10 @@ OperatorResultType PhysicalNestedLoopJoin::ExecuteInternal(ExecutionContext &con
 	}
 
 	switch (join_type) {
-	case JoinType::SEMI:
-	case JoinType::ANTI:
+	case JoinType::LEFT_SEMI:
+	case JoinType::LEFT_ANTI:
+	case JoinType::RIGHT_SEMI:
+	case JoinType::RIGHT_ANTI:
```

Comment:
It's a bit misleading to add `RIGHT_SEMI` and `RIGHT_ANTI` to the switch here because it will throw a `NotImplementedException` later anyway. I would remove these cases